src/main/java/de/tum/cit/aet/artemis/exercise/repository/StudentParticipationRepository.java

```diff
@@ -1199,51 +1199,121 @@ SELECT COALESCE(AVG(p.presentationScore), 0)
+                         SELECT MAX(pr.id)
+                         FROM p.results pr
+                         WHERE pr.participation.id = p.id
+                     )
                  JOIN r.feedbacks f
-                    WHERE p.exercise.id = :exerciseId
-                        AND p.testRun = FALSE
-                        AND r.id = (
-                            SELECT MAX(pr.id)
-                            FROM p.results pr
-                            )
-                        AND f.positive = FALSE
-                        GROUP BY f.detailText, f.testCase.testName
+            WHERE p.exercise.id = :exerciseId
+                  AND p.testRun = FALSE
+                  AND f.positive = FALSE
+                  AND r.id = (
+                      SELECT MAX(pr.id)
+                      FROM p.results pr
+                  )
+                  AND (:searchTerm = '' OR LOWER(f.detailText) LIKE LOWER(CONCAT('%', :searchTerm, '%')))
+                  AND (:filterTestCases IS NULL OR f.testCase.testName IN (:filterTestCases))
+                  AND (:filterTaskNames IS NULL OR f.testCase.testName IN (
+                      SELECT tct.testName
+                      FROM ProgrammingExerciseTask t
+                      JOIN t.testCases tct
+                      WHERE t.taskName IN (:filterTaskNames)
+                  ))
+            GROUP BY f.detailText, f.testCase.testName
+            HAVING COUNT(f.id) BETWEEN :minOccurrence AND :maxOccurrence
             """)
-    List<FeedbackDetailDTO> findAggregatedFeedbackByExerciseId(@Param("exerciseId") long exerciseId);
+    Page<FeedbackDetailDTO> findFilteredFeedbackByExerciseId(@Param("exerciseId") long exerciseId, @Param("searchTerm") String searchTerm,
+            @Param("filterTestCases") List<String> filterTestCases, @Param("filterTaskNames") List<String> filterTaskNames, @Param("minOccurrence") long minOccurrence,
+            @Param("maxOccurrence") long maxOccurrence, Pageable pageable);
 
     /**
      * Counts the distinct number of latest results for a given exercise, excluding those in practice mode.
+     * <br>
+     * For each participation, it selects only the latest result (using MAX) and ensures that the participation is not a test run.
      *
-     * @param exerciseId Exercise ID.
-     * @return The count of distinct latest results for the exercise.
+     * @param exerciseId Exercise ID for which distinct results should be counted.
+     * @return The total number of distinct latest results for the given exercise.
      */
     @Query("""
-                SELECT COUNT(DISTINCT r.id)
+            SELECT COUNT(DISTINCT r.id)
+            FROM StudentParticipation p
+                JOIN p.results r
+            WHERE p.exercise.id = :exerciseId
+                  AND p.testRun = FALSE
+                  AND r.id = (
+                      SELECT MAX(pr.id)
+                      FROM p.results pr
+                  )
```

Comment:
Here you do this check in the where clause but in the next query you do it in the join clause. Can you unify this (preferably in the join clause)?